libfreerdp/core/transport.c

```diff
@@ -1079,22 +1084,25 @@ static int transport_default_read_pdu(rdpTransport* transport, wStream* s)
 	{
 		if (!Stream_EnsureCapacity(s, 4))
 			return -1;
-		const int rc = transport_read_layer_bytes(transport, s, 4);
+		const SSIZE_T rc = transport_read_layer_bytes(transport, s, 4);
+		if (rc > INT32_MAX)
```

Comment:
`transport_read_layer_bytes` is supposed to just return -1, 0 or 1 so anything else should be considered an error

Reply:
good point, adjusted